substrate/frame/support/procedural/src/construct_runtime/expand/origin.rs

```diff
@@ -153,6 +153,10 @@ pub fn expand_outer_origin(
 				self.filter = #scrate::__private::sp_std::rc::Rc::new(Box::new(filter));
 			}
 
+			fn set_caller(&mut self, caller: OriginCaller) {
```

Comment:
I suppose this goes back to the new tx extensions being able to alter origin?

Reply:
If this is the killer feature, it must be documented in `reference_docs::transaction_extensions`.

Reply:
This enables `TransactionExtension`s to alter origins during validation/preparation. This should be documented in an example of new-school transaction impl using `TransactionExtension`. Opened #3593 to address this.